src/endpoint/scheduler.rs

```diff
@@ -370,6 +374,36 @@ impl JobHandle {
     }
 }
 
+impl Drop for JobHandle {
+    fn drop(&mut self) {
+        debug!("Cleaning up JobHandle");
+        if self.container_id.is_some() {
+            debug!("Container was already started");
+            let docker = self.endpoint.docker().clone();
+            let container_id = self.container_id.take().unwrap();
```

Comment:
This isn't pretty and not guaranteed to be safe - please use `if let Some(container_id) = self.container_id` or something similar instead.

Reply:
I tried that. It didn't work because of the lifetimes

Reply:
That most likely was due to something else then. I can give it a look after my vacation but I don't see why that should create (unsolvable) issues.

Reply:
You can use the following patch:
```diff
--- a/src/endpoint/scheduler.rs
+++ b/src/endpoint/scheduler.rs
@@ -377,10 +377,9 @@ impl JobHandle {
 impl Drop for JobHandle {
     fn drop(&mut self) {
         debug!("Cleaning up JobHandle");
-        if self.container_id.is_some() {
+        if let Some(container_id) = self.container_id.clone() {
             debug!("Container was already started");
             let docker = self.endpoint.docker().clone();
-            let container_id = self.container_id.take().unwrap();

             tokio::spawn(async move {
                 let container = docker.containers().get(&container_id);
```